gen/src/main/java/com/softlayer/api/gen/ClassWriter.java

```diff
@@ -377,8 +377,14 @@ public ClassWriter emitType() throws IOException {
         
         // Now the service
         if (!type.meta.noservice) {
-            if (type.meta.properties.containsKey("id")) {
-                if (type.meta.properties.containsKey("globalIdentifier")) {
+
+            Boolean containsId = type.meta.properties.containsKey("id")
+                    || type.baseMeta.properties.containsKey("id");
```

Comment:
I've added the reference to `Meta` for access within `ClassWriter` and it now traverses down though the parent `Meta.Type`s to determine if `id` or `globalIdentifier` properties exist.